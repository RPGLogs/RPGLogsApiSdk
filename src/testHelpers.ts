import { buildSdk } from './main';
import btoa from 'btoa';

export function buildTestSdk(): ReturnType<typeof buildSdk> {
  return buildSdk(<string>process.env.TEST_ACCESS_TOKEN);
}

export async function getAccessToken(
  clientId: string,
  clientSecret: string,
  game = 'warcraft'
): Promise<string> {
  if (game === 'warcraft') game = 'www.warcraft';

  const authHeader = 'Basic ' + btoa(clientId + ':' + clientSecret);

  const response = await fetch(`https://${game}logs.com/oauth/token`, {
    method: 'POST',
    headers: {
      Authorization: authHeader,
      'Content-Type': 'application/x-www-form-urlencoded',
    },
    body: 'grant_type=client_credentials',
  });

  const json = await response.json();

  if (response.status === 200) {
    return json.access_token;
  } else {
    throw new Error('Response was not OK: ' + JSON.stringify(json ?? {}));
  }
}
