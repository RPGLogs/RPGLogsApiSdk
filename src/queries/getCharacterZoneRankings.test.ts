import { buildTestSdk } from '../testHelpers';

test('returns data', async () => {
  const response = await buildTestSdk().getCharacterZoneRankings({
    zoneId: 24,
    characterId: 46159040,
  });

  expect(response?.characterData?.character).toMatchSnapshot();
});
