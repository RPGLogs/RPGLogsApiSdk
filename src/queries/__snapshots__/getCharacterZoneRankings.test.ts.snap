// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns data 1`] = `
{
  "id": 46159040,
  "name": "Xy√§",
  "zoneRankings": {
    "allStars": [
      {
        "partition": 3,
        "points": 963.45,
        "possiblePoints": 1080,
        "rank": 52,
        "rankPercent": 99.61331412540754,
        "regionRank": 21,
        "serverRank": 52,
        "spec": "Vengeance",
        "total": 13189,
      },
      {
        "partition": 2,
        "points": 87.4102,
        "possiblePoints": 1320,
        "rank": 4815,
        "rankPercent": 45.27057753524329,
        "regionRank": 2143,
        "serverRank": null,
        "spec": "Vengeance",
        "total": 8796,
      },
      {
        "partition": 3,
        "points": 111.307,
        "possiblePoints": 1080,
        "rank": 36951,
        "rankPercent": 44.707977314558484,
        "regionRank": 14523,
        "serverRank": 36951,
        "spec": "Havoc",
        "total": 66827,
      },
    ],
    "bestPerformanceAverage": 93.49076666666667,
    "difficulty": 5,
    "medianPerformanceAverage": 65.55116666666666,
    "metric": "dps",
    "partition": -1,
    "rankings": [
      {
        "allStars": {
          "partition": 3,
          "points": 115.92,
          "possiblePoints": 120,
          "rank": 30,
          "rankPercent": 99.70244202749846,
          "regionRank": 9,
          "serverRank": 30,
          "total": 9746,
        },
        "bestAmount": 69568.1,
        "bestSpec": "Vengeance",
        "encounter": {
          "id": 2329,
          "name": "Wrathion",
        },
        "fastestKill": 127266,
        "lockedIn": true,
        "medianPercent": 85.3273,
        "rankPercent": 99.3903,
        "spec": "Vengeance",
        "totalKills": 20,
      },
      {
        "allStars": {
          "partition": 3,
          "points": 115.21,
          "possiblePoints": 120,
          "rank": 12,
          "rankPercent": 99.87502840263576,
          "regionRank": 5,
          "serverRank": 12,
          "total": 8802,
        },
        "bestAmount": 91289.5,
        "bestSpec": "Vengeance",
        "encounter": {
          "id": 2327,
          "name": "Maut",
        },
        "fastestKill": 82616,
        "lockedIn": true,
        "medianPercent": 82.1239,
        "rankPercent": 99.4233,
        "spec": "Vengeance",
        "totalKills": 18,
      },
      {
        "allStars": {
          "partition": 3,
          "points": 0,
          "possiblePoints": 0,
          "rank": 443,
          "rankPercent": 95.46713157624859,
          "regionRank": 185,
          "serverRank": 443,
          "total": 9751,
        },
        "bestAmount": 49920.8,
        "bestSpec": "Vengeance",
        "encounter": {
          "id": 2334,
          "name": "Prophet Skitra",
        },
        "fastestKill": 99050,
        "lockedIn": true,
        "medianPercent": 78.3728,
        "rankPercent": 95.2483,
        "spec": "Vengeance",
        "totalKills": 19,
      },
      {
        "allStars": {
          "partition": 3,
          "points": 111.08,
          "possiblePoints": 120,
          "rank": 103,
          "rankPercent": 96.64252797893351,
          "regionRank": 45,
          "serverRank": 103,
          "total": 3038,
        },
        "bestAmount": 71138.5,
        "bestSpec": "Vengeance",
        "encounter": {
          "id": 2328,
          "name": "Dark Inquisitor Xanesh",
        },
        "fastestKill": 147415,
        "lockedIn": true,
        "medianPercent": 59.158,
        "rankPercent": 96.5564,
        "spec": "Vengeance",
        "totalKills": 14,
      },
      {
        "allStars": {
          "partition": 3,
          "points": 0,
          "possiblePoints": 0,
          "rank": 1,
          "rankPercent": 100,
          "regionRank": 1,
          "serverRank": 1,
          "total": 4977,
        },
        "bestAmount": 255365,
        "bestSpec": "Vengeance",
        "encounter": {
          "id": 2333,
          "name": "The Hivemind",
        },
        "fastestKill": 79259,
        "lockedIn": true,
        "medianPercent": 71.4902,
        "rankPercent": 100,
        "spec": "Vengeance",
        "totalKills": 19,
      },
      {
        "allStars": {
          "partition": 3,
          "points": 102.5,
          "possiblePoints": 120,
          "rank": 367,
          "rankPercent": 91.99124726477024,
          "regionRank": 161,
          "serverRank": 367,
          "total": 4570,
        },
        "bestAmount": 62138.2,
        "bestSpec": "Vengeance",
        "encounter": {
          "id": 2335,
          "name": "Shad'har the Insatiable",
        },
        "fastestKill": 135857,
        "lockedIn": true,
        "medianPercent": 69.9785,
        "rankPercent": 91.7203,
        "spec": "Vengeance",
        "totalKills": 18,
      },
      {
        "allStars": {
          "partition": 3,
          "points": 0,
          "possiblePoints": 0,
          "rank": 25,
          "rankPercent": 99.07904834996162,
          "regionRank": 10,
          "serverRank": 25,
          "total": 2606,
        },
        "bestAmount": 75561,
        "bestSpec": "Vengeance",
        "encounter": {
          "id": 2343,
          "name": "Drest'agath",
        },
        "fastestKill": 173027,
        "lockedIn": true,
        "medianPercent": 84.6782,
        "rankPercent": 99.0088,
        "spec": "Vengeance",
        "totalKills": 17,
      },
      {
        "allStars": {
          "partition": 3,
          "points": 111.49,
          "possiblePoints": 120,
          "rank": 8,
          "rankPercent": 99.75786924939467,
          "regionRank": 7,
          "serverRank": 8,
          "total": 2891,
        },
        "bestAmount": 115720,
        "bestSpec": "Vengeance",
        "encounter": {
          "id": 2336,
          "name": "Vexiona",
        },
        "fastestKill": 127265,
        "lockedIn": true,
        "medianPercent": 82.5465,
        "rankPercent": 99.4418,
        "spec": "Vengeance",
        "totalKills": 14,
      },
      {
        "allStars": {
          "partition": 3,
          "points": 105.65,
          "possiblePoints": 120,
          "rank": 217,
          "rankPercent": 94.42724458204334,
          "regionRank": 95,
          "serverRank": 217,
          "total": 3876,
        },
        "bestAmount": 79258.7,
        "bestSpec": "Vengeance",
        "encounter": {
          "id": 2331,
          "name": "Ra-den the Despoiled",
        },
        "fastestKill": 114571,
        "lockedIn": true,
        "medianPercent": 62.6349,
        "rankPercent": 95.0572,
        "spec": "Vengeance",
        "totalKills": 18,
      },
      {
        "allStars": {
          "partition": 3,
          "points": 114.06,
          "possiblePoints": 120,
          "rank": 36,
          "rankPercent": 98.22964087000506,
          "regionRank": 13,
          "serverRank": 36,
          "total": 1977,
        },
        "bestAmount": 69776.1,
        "bestSpec": "Vengeance",
        "encounter": {
          "id": 2345,
          "name": "Il'gynoth, Corruption Reborn",
        },
        "fastestKill": 310516,
        "lockedIn": true,
        "medianPercent": 65.2973,
        "rankPercent": 96.9106,
        "spec": "Vengeance",
        "totalKills": 15,
      },
      {
        "allStars": {
          "partition": 3,
          "points": 111.17,
          "possiblePoints": 120,
          "rank": 46,
          "rankPercent": 97.38372093023256,
          "regionRank": 16,
          "serverRank": 46,
          "total": 1720,
        },
        "bestAmount": 60696.7,
        "bestSpec": "Vengeance",
        "encounter": {
          "id": 2337,
          "name": "Carapace of N'Zoth",
        },
        "fastestKill": 299254,
        "lockedIn": true,
        "medianPercent": 41.177,
        "rankPercent": 97.3633,
        "spec": "Vengeance",
        "totalKills": 15,
      },
      {
        "allStars": {
          "partition": 3,
          "points": 76.37,
          "possiblePoints": 120,
          "rank": 375,
          "rankPercent": 64.38095238095238,
          "regionRank": 168,
          "serverRank": 375,
          "total": 1050,
        },
        "bestAmount": 57766.1,
        "bestSpec": "Vengeance",
        "encounter": {
          "id": 2344,
          "name": "N'Zoth the Corruptor",
        },
        "fastestKill": 523007,
        "lockedIn": true,
        "medianPercent": 41.7171,
        "rankPercent": 65.5537,
        "spec": "Vengeance",
        "totalKills": 10,
      },
    ],
    "zone": 24,
  },
}
`;
